python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/utils/anthropic/_attributes.py

```diff
@@ -0,0 +1,178 @@
+                    )
+                    messages.append(tool_call_result)
+                else:
+                    contents.append(TextMessageContent(text=part_content.get("text"), type="text"))
+            if contents or tool_calls:
+                messages.append(
+                    Message(contents=contents, tool_calls=tool_calls, role=message.get("role"))
+                )
+    return messages
+
+
+def _get_llm_token_counts(usage: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
+    prompt_tokens = sum(
+        usage.get(key, 0)
+        for key in [
+            "input_tokens",
+            "cache_creation_input_tokens",
+            "cache_read_input_tokens",
+        ]
+    )
+    if prompt_tokens:
+        yield LLM_TOKEN_COUNT_PROMPT, prompt_tokens
+
+    if usage.get("output_tokens"):
+        yield LLM_TOKEN_COUNT_COMPLETION, usage.get("output_tokens")
+    if usage.get("cache_read_input_tokens"):
+        yield LLM_TOKEN_COUNT_PROMPT_DETAILS_CACHE_READ, usage.get("cache_read_input_tokens")
+    if usage.get("cache_creation_input_tokens"):
+        yield LLM_TOKEN_COUNT_PROMPT_DETAILS_CACHE_WRITE, usage.get("cache_creation_input_tokens")
+
+
+def _get_llm_model_name_from_input(model_id: str) -> str:
+    if model_id and model_id.count(".") == 1:
+        vendor, model_name = model_id.split(".")
+        return model_name
+    return model_id
+
+
+def _get_invocation_parameters(kwargs: Mapping[str, Any]) -> Any:
+    """
+    Extracts the invocation parameters from the call
+    """
+    return {key: value for key, value in kwargs.items() if _validate_invocation_parameter(key)}
+
+
+def _get_output_messages(response: Any) -> Any:
+    """
+    Extracts the tool call information from the response
+    """
+    output_messages = []
+    for block in response.get("content", []):
+        if block.get("type") == "tool_use":
+            tool_call_function = ToolCallFunction(
+                name=block.get("name", ""), arguments=block.get("input")
+            )
+            tool_calls = [ToolCall(id=block.get("id", ""), function=tool_call_function)]
+            output_messages.append(
+                Message(
+                    tool_call_id=block.get("tool_call_id", ""),
+                    role=response.get("role"),
+                    tool_calls=tool_calls,
+                )
+            )
+
+        if block.get("type") == "text":
+            output_messages.append(
+                Message(content=block.get("text"), role=response.get("role") or "assistant")
+            )
+    return output_messages
+
+
+def _validate_invocation_parameter(parameter: Any) -> bool:
+    """
+    Validates the invocation parameters.
+    """
+    valid_params = (
+        "max_tokens",
+        "max_tokens_to_sample",
+        "model",
+        "metadata",
+        "stop_sequences",
+        "stream",
+        "system",
+        "temperature",
+        "tool_choice",
+        "tools",
+        "top_k",
+        "top_p",
+    )
```

Comment:
this seems to be a bit of the inverse of what we want as parameters keep getting added like reasoning budget. Can we instead keep a list of params that are invalid?